src/main/java/com/fulcrumgenomics/util/PickLongIndices.scala

```diff
@@ -0,0 +1,291 @@
+  @arg(flag="g", doc="The minimum GC fraction for a barcode to be accepted.")          val minGc: Double = 0.2,
+  @arg(flag="G", doc="The maximum GC fraction for a barcode to be accepted.")          val maxGc: Double = 0.8,
+  @arg(          doc="File of existing index sequences to integrate, one per line.")   val existing: Option[FilePath] = None,
+  @arg(flag="s", doc="Random seed value.")                                             val seed: Int = 1,
+  @arg(flag="a", doc="Attempts to pick the next index before quitting.")               val attempts: Int = 1e6.toInt,
+  @arg(          doc="The installation directory for ViennaRNA.")                      val viennaRnaDir: Option[DirPath] = None,
+  @arg(flag="t", doc="The temperature at which to predict secondary structure.")       val temperature: Double = 25d,
+  @arg(          doc="The lowest acceptable secondary structure deltaG.")              val minDeltaG: Double = -10,
+  @arg(          doc="Adapter sequence(s) indices will be inserted into.", minElements=0) val adapters: Seq[String] = Seq(),
+  @arg(          doc="Sequences that should be avoided.  Any kmer of 'length' that appears in these sequences and their " + "reverse complements will be thrown out.")
+  val avoidSequence: Seq[String] = IlluminaAdapters.all.flatMap(_.both)
+) extends FgBioTool with LazyLogging {
+  // A few "constants"
+  private val Bases  = Array[Byte]('A', 'C', 'G', 'T')
+  private val random = new Random(seed)
+
+  // Input checking
+  existing.foreach(Io.assertReadable)
+  Io.assertCanWriteFile(output)
+  if (length < 1)                invalid("Length must be a positive integer.")
+  if (numberOfIndices < 1)       invalid("Number of indices to pick must be a positive integer.")
+  if (maxHomopolymer < 0)        invalid("Max homopolymer length must be >= 0.")
+  if (minGc >= maxGc)            invalid("Min GC must be greater than max GC.")
+  if (editDistance < 0)          invalid("Edit distance must be >= 0.")
+  if (editDistance > length)     invalid("Edit distance must be <= length.")
+  if (viennaRnaDir.isDefined && adapters.isEmpty) invalid("Specifying Vienna RNA dir has no effect if no adapters are provided.")
+  if (viennaRnaDir.isEmpty && adapters.nonEmpty)  invalid("Specifying adapters has no effect if Vienna RNA dir is not provided.")
+
+  /** Parse and check any set of existing indices. */
+  private val ExistingIndices = existing match {
+    case None    => Seq.empty[Index]
+    case Some(p) => Io.readLines(p).map(_.trim.toUpperCase).filter(_.nonEmpty).map(_.getBytes).toSeq
+  }
+
+  if (ExistingIndices.exists(_.length != length)) invalid(s"Existing indices are not all of length ${length}")
+  if (ExistingIndices.exists(idx => idx.exists(b => !Bases.contains(b)))) invalid("One or more existing indices contains a non-[ACGT] character.")
+
+  /** The set of KMERs from the avoid sequences, that should not be selected. */
+  private val AvoidKmers = avoidSequence.flatMap { seq =>
```

Comment:
I would change them to lower case for style purposes.  They are vals so they are constants.  Either that or move them to an object.